microsoft/testsuites/core/storage.py

```diff
@@ -294,21 +300,34 @@ def verify_os_partition_identifier(self, log: Logger, node: RemoteNode) -> None:
         )
         os_partition_info = node.tools[Blkid].get_partition_info_by_name(os_partition)
 
+        # check if cvm
+        is_cvm = False
+        settings = Feature.get_feature_settings(
+            node.features[SecurityProfile]._settings
```

Comment:
1. Avoid to use feature to check if it's CVM or not. Is there a way to check it from the VM itself?
2. The _settings is a private method. Add a method in SecurityProfile feature to return the value.

Reply:
1. I checked with Karmeron there's no existing mechanism in LISA to determine if it's a CVM from the VM itself.
2. will update

Reply:
> * I checked with Karmeron there's no existing mechanism in LISA to determine if it's a CVM from the VM itself.

Do you mean not exists in LISA or in the VM? If it's not in LISA, but in VM, we can add it.

Reply:
I did some search and try and it seems not feasible to directly get this info from VM itself:
1. there's no info from dmesg/journalctl clearly indicates this is CVM;
2. the image/SKU name may contain keywords like 'confidential' but it looks not safe to me
3. az cli can get the vm metadata which contains security type to show if it's cvm or not, but this metadata seems to be stored on azure side not the vm itself, a similar command 'curl -H Metadata:true' running from vm only shows secureBootEnabled and virtualTpmEnabled info but no securityType

Reply:
Thanks a lot for a deep investigation.